controllers/apps/transformer_component_service.go

```diff
@@ -89,111 +71,46 @@ func (t *componentServiceTransformer) Transform(ctx graph.TransformContext, dag
 	return nil
 }
 
-func (t *componentServiceTransformer) genMultiServicesIfNeed(cluster *appsv1alpha1.Cluster,
-	synthesizeComp *component.SynthesizedComponent, compService *appsv1alpha1.ComponentService) ([]*appsv1alpha1.ComponentService, error) {
-	if !compService.GeneratePodOrdinalService {
-		return []*appsv1alpha1.ComponentService{compService}, nil
+func (t *componentServiceTransformer) buildCompService(comp *appsv1alpha1.Component,
+	synthesizeComp *component.SynthesizedComponent, service *appsv1alpha1.ComponentService) ([]*corev1.Service, error) {
+	if service.DisableAutoProvision != nil && *service.DisableAutoProvision {
+		return nil, nil
 	}
 
-	podOrdinalServices := make([]*appsv1alpha1.ComponentService, 0, synthesizeComp.Replicas)
+	if service.PodService == nil || !*service.PodService {
+		return t.buildServices(comp, synthesizeComp, []*appsv1alpha1.ComponentService{service})
+	}
+
+	services := make([]*appsv1alpha1.ComponentService, 0)
 	for i := int32(0); i < synthesizeComp.Replicas; i++ {
-		svc := compService.DeepCopy()
-		svc.Name = fmt.Sprintf("%s-%d", compService.Name, i)
-		if len(compService.ServiceName) == 0 {
+		svc := service.DeepCopy()
+		svc.Name = fmt.Sprintf("%s-%d", service.Name, i)
+		if len(service.ServiceName) == 0 {
 			svc.ServiceName = fmt.Sprintf("%d", i)
 		} else {
-			svc.ServiceName = fmt.Sprintf("%s-%d", compService.ServiceName, i)
+			svc.ServiceName = fmt.Sprintf("%s-%d", service.ServiceName, i)
 		}
 		if svc.Spec.Selector == nil {
 			svc.Spec.Selector = make(map[string]string)
 		}
 		// TODO(xingran): use StatefulSet's podName as default selector to select unique pod
-		svc.Spec.Selector[constant.StatefulSetPodNameLabelKey] = constant.GeneratePodName(cluster.Name, synthesizeComp.Name, int(i))
-		podOrdinalServices = append(podOrdinalServices, svc)
+		svc.Spec.Selector[constant.StatefulSetPodNameLabelKey] = constant.GeneratePodName(synthesizeComp.ClusterName, synthesizeComp.Name, int(i))
```

Comment:
in KubeBlocks v0.9.0,  the dependency on StatefulSet has been removed, so the StatefulSetPodNameLabelKey is no longer applicable here. A new key needs to be used as a replacement (which may not exist yet).

Reply:
Will submit a separate PR to solve this problem